packages/p2p/src/pages/my-ads/order-time-selection/order-time-selection.tsx

```diff
@@ -23,28 +25,31 @@ const OrderTimeSelection = ({
     ...field
 }: TOrderTimeSelectionProps) => {
     const { values, handleChange }: TFormikContext = useFormikContext<TFormikContext>();
+    const { order_completion_time } = values;
     const { showModal } = useModalManagerContext();
+    const { p2p_settings } = useP2PSettings();
+    const { order_expiry_options } = p2p_settings ?? {};
     const { ui } = useStore();
     const { is_mobile } = ui;
     const order_time_info_message = localize('Orders will expire if they aren’t completed within this time.');
-    const order_completion_time_list = [
-        {
-            text: localize('1 hour'),
-            value: '3600',
-        },
-        {
-            text: localize('45 minutes'),
-            value: '2700',
-        },
-        {
-            text: localize('30 minutes'),
-            value: '1800',
-        },
-        {
-            text: localize('15 minutes'),
-            value: '900',
-        },
-    ];
+
+    const getOrderExpiryOptions = (
+        order_expiry_options: NonNullable<
+            NonNullable<ReturnType<typeof useP2PSettings>>['p2p_settings']
+        >['order_expiry_options']
+    ) => {
+        const options = order_expiry_options?.map(option => ({
+            text: formatTime(option / 60),
+            value: `${option}`,
+        }));
+        if (options?.some(option => option.value === order_completion_time)) return options;
+        return (
+            options?.concat({
+                text: formatTime(Number(order_completion_time) / 60),
+                value: order_completion_time,
+            }) ?? []
+        );
```

Comment:
what is this function for?

Reply:
in the case of edit, if user has previously selected, 30 mins as the order expiry time and then if that 30 mins was removed from the list of options, we have to manually add the 30 mins option. so this checks if the value exists and adds it if needed, or else returns the list of options.